tasks/src/test/java/taf/page/YandexDiskMainPage.java

```diff
@@ -0,0 +1,153 @@
+package taf.page;
+
+import static taf.browser.Browser.*;
+import static taf.constants.Constants.*;
+
+import java.util.Random;
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+
+public class YandexDiskMainPage extends YandexDiskAbstractPage {
+  private static String folderName;
+  private WebElement folderNameInput;
+
+  private String generateFolderName() {
+    Random random = new Random();
+    return "folder_" + random.nextInt(RANDOM_NUMBER_BOUND);
+  }
+
+  public YandexDiskMainPage(WebDriver driver) {
+    super(driver);
+  }
+
+  public String getNewestLink() {
+    waitElementToBeClickable(driver, NEWEST_TITLE);
+    return driver.findElement(NEWEST_TITLE).getAttribute("href");
+  }
+
+  public String getFilesLink() {
+    return driver.findElement(ID_FILES).getAttribute("href");
+  }
+
+  public String getPhotoLink() {
+    return driver.findElement(PHOTO_TITLE).getAttribute("href");
+  }
+
+  public String getPublishedLink() {
+    return driver.findElement(SHARED_ACCESS_TITLE).getAttribute("href");
+  }
+
+  public String getHistoryLink() {
+    return driver.findElement(HISTORY_TITLE).getAttribute("href");
+  }
+
+  public String getArchiveLink() {
+    return driver.findElement(ARCHIVE_TITLE).getAttribute("href");
+  }
+
+  public String getTrashLink() {
+    return driver.findElement(ID_TRASH).getAttribute("href");
+  }
+
+  public YandexDiskMainPage clickMenuItemFiles() {
+    waitElementToBeClickable(driver, ID_FILES);
+    driver.findElement(ID_FILES).click();
+    return this;
+  }
+
+  public YandexDiskMainPage createFolder() {
+    waitElementToBeClickable(driver, CREATE_RESOURCE);
+    driver.findElement(CREATE_RESOURCE).click();
+
+    waitElementToBeClickable(driver, CREATE_RESOURCE_FOLDER);
+    driver.findElement(CREATE_RESOURCE_FOLDER).click();
+    return this;
+  }
+
+  public YandexDiskMainPage pressCreateFolder() {
+    folderNameInput.sendKeys(Keys.ENTER);
+    return this;
+  }
+
+  public YandexDiskMainPage typeFolderName() {
+    folderName = generateFolderName();
+
+    folderNameInput = driver.findElement(FOLDER_NAME_INPUT);
+    folderNameInput.sendKeys(Keys.CONTROL + "a");
+    waitForVisibleElement(driver, folderNameInput);
+    folderNameInput.sendKeys(folderName);
+    return this;
+  }
+
+  public YandexDiskMainPage visitCreatedFolder() {
+    Actions actions = new Actions(driver);
+    By createdFolder = By.xpath("//span[contains(text(), '"+ folderName +"')]/parent::div");
```

Comment:
Please do not hardcode locators it's not clear what is the exact element. Create a class field with a descriptive name.